src/sampler.jl

```diff
@@ -40,8 +40,9 @@ function sample_init(
 )
     # Ensure h.metric has the same dim as θ.
     h = resize(h, θ)
+    refresh_r = rand_momentum(rng, h.metric, h.kinetic, θ) # Momentum refreshment
     # Initial transition
-    t = Transition(phasepoint(rng, θ, h), NamedTuple())
+    t = Transition(phasepoint(h, θ, refresh_r), NamedTuple())
```

Comment:
```suggestion
    t = Transition(phasepoint(h, θ, refreshed_r), NamedTuple())
```

and similarly, all other places.